picard/tagger.py

```diff
@@ -658,9 +666,14 @@ def get_files_from_objects(self, objects, save=False):
 
     def save(self, objects):
         """Save the specified objects."""
-        files = self.get_files_from_objects(objects, save=True)
-        for file in files:
-            file.save()
+        for o in objects:
+            log.debug('save object %r', o)
+            files = self.get_files_from_objects([o, ], save=True)
```

Comment:
This defeats uniquification (?) implemented in https://github.com/metabrainz/picard/blob/7b41e1283b4fd740e1fe4fb3933961c72c4d489f/picard/tagger.py#L663-L665

Reply:
Yes, this can potentially lead to the same file saved twice: You can have both the album and the files selected for saving, but saving should only happen once.

Also we need to be clear what is expected to happen:

1. If I select the album and save, should the album_save_post_processor be run? (I assume this is a clear YES)
2. If I select all files of an album, but not the album itself, should the album_save_post_processor be run?
3. If I select only some files of an album, but not the album itself, should the album_save_post_processor be run?

I tend to say yes to all of the above. But for case 3 it might be useful if the registered processor gets some information, that only a subset has been saved. Optimal would be to pass a list of actually saved files to the processor, so it could figure out what to do with this based on the use case.